predicators/approaches/grammar_search_invention_approach.py

```diff
@@ -471,20 +518,62 @@ def enumerate(self) -> Iterator[Tuple[Predicate, float]]:
                     # learn a classifier with it. So, skip the feature.
                     if abs(lb - ub) < 1e-6:
                         continue
-                    # Scale the constant by the feature range.
-                    k = constant * (ub - lb) + lb
-                    # Only need one of (ge, le) because we can use negations
-                    # to get the other (modulo equality, which we shouldn't
-                    # rely on anyway because of precision issues).
-                    comp, comp_str = le, "<="
-                    classifier = _SingleAttributeCompareClassifier(
-                        0, t, f, k, constant_idx, comp, comp_str)
-                    name = str(classifier)
-                    types = [t]
-                    pred = Predicate(name, types, classifier)
-                    assert pred.arity == 1
+                    pred = self._construct_unary_pred(t, f, constant, ub, lb,
+                                                      constant_idx)
                     yield (pred, 1 + cost)  # cost = arity + cost from constant
 
+    def _enumerate_transition_changes(
+        self, feature_ranges: Dict[Type, Dict[str, Tuple[float, float]]]
+    ) -> Iterator[Tuple[Predicate, float]]:
+        constant_generator = _halving_constant_generator(0.0, 1.0)
+        constant_idx = 0
+        while True:
+            # First generate a set of constants that all predicates in
+            # this iteration will be evaluated on.
+            curr_iter_constants_and_costs = []
+            for _ in range(CFG.grammar_search_num_constants_each_iter):
+                try:
+                    curr_iter_constants_and_costs.append(
+                        next(constant_generator))
+                except StopIteration:  # pragma: no cover
```

Comment:
if this is impossible, why do we need to catch it?

Reply:
if we don't, `pylint` starts to complain